### DISCUSSION TITLE
Disable global styles

### DISCUSSION THREAD

**Discussion by CypherPotato** [spectreconsole/spectre.console#1673] [Q&A] [answered]
Is there any option to globally disable styles for `AnsiConsole` without creating an interceptor for that? I want to give my program users an option to get raw, not formatted output, but I doens't want to adapt my code to receive two formats of lines.

Currently, the only way I know is:

```vb.net
If isColorsEnabled Then
    AnsiConsole.MarkupLine("[bold red on black]Hello, world!![/]")
Else
    Console.WriteLine("Hello, world!")
End If
```

That's two different strings, and I have to change across multiple files that uses `AnsiConsole`.

**Comment by nils-a** [comment, marked as answer]
If you only want the colors disabled, spectre.console supports [NO_COLOR](https://no-color.org). Simply setting an env variable `NO_COLOR=1` should remove all colors from the output.

If you need a bit more control, you can change the profile of `AnsiConsole` - you could do that once and all output will be changed.

For example, like this:
```cs
// remove all colors from the output
AnsiConsole.Console.Profile.Capabilities.ColorSystem = ColorSystem.NoColors;

// remove all ANSI sequences 
AnsiConsole.Console.Profile.Capabilities.Ansi = false;
```
